packages/typespec-ts/src/modular/buildProjectFiles.ts

```diff
@@ -85,34 +84,21 @@ function buildExportsForMultiClient(
   return packageInfo;
 }
 
-function buildPackageDetails(packageDetails?: PackageDetails) {
-  return (
-    packageDetails ?? {
-      name: "@msinternal/unamedpackage",
-      nameWithoutScope: "unamedpackage",
-      version: "1.0.0-beta.1"
-    }
-  );
-}
-
 // Prepare package info without scripts and devDependencies and dependencies ect
 function initPackageInfo(codeModel: ModularCodeModel) {
-  let { packageDetails, generateTest, branded } = codeModel.options;
-  branded = branded ?? true;
-  generateTest = generateTest === true || generateTest === undefined;
-  packageDetails = buildPackageDetails(packageDetails);
-  const description = packageDetails.description
-    ? packageDetails.description
+  const { packageDetails, generateTest, branded } = codeModel.options;
+  const description = packageDetails!.description
```

Comment:
could we add an assertion not use ! here? maybe throw execption if undefined.

Reply:
technically packageDetails could not be undefined, as above https://github.com/Azure/autorest.typescript/pull/2296#discussion_r1498694204 but we can't define it as a required emitter option.